src/lib/build.js

```diff
@@ -1,14 +1,34 @@
 const build = require('@netlify/build')
 
+const { getSiteInformation } = require('../utils/dev')
+
+const getBuildEnv = async ({ context }) => {
+  const { warn, error, netlify } = context
+  const { site, api } = netlify
+  const { teamEnv, addonsEnv, siteEnv, dotFilesEnv } = await getSiteInformation({
+    api,
+    site,
+    warn,
+    error,
+  })
+  const env = { ...teamEnv, ...addonsEnv, ...siteEnv, ...dotFilesEnv }
```

Comment:
@ehmicky  since `@netlify/config` already evaluates some of these, should we move this logic to `@netlify/config`?

Reply:
Yes.
I have filed https://github.com/netlify/build/issues/2036 with additional information. However, this can be done in a follow-up PR.